pkg/bff/credentials.go

```diff
@@ -0,0 +1,150 @@
+package bff
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	jwt "github.com/golang-jwt/jwt/v4"
+	"github.com/trisacrypto/directory/pkg/bff/config"
+	"github.com/trisacrypto/directory/pkg/gds/admin/v2"
+	"github.com/trisacrypto/directory/pkg/gds/tokens"
+)
+
+// Credentials implements the admin.Credentials interface to provide access and refresh
+// tokens to the GDSClient.
+type Credentials struct {
+	tm      *tokens.TokenManager
+	access  string
+	refresh string
+}
+
+// NewCredentials creates a new Credentials object which can generate new access and
+// refresh tokens for authentication based on the provided token keys and audience.
+func NewCredentials(conf config.AdminConfig) (creds *Credentials, err error) {
```

Comment:
Could we also change this so that it returns an `admin.Credentials`? This is just another helpful indication that this thing is implementing the credentials interface.